MediaBrowser.Controller/MediaEncoding/EncodingHelper.cs

```diff
@@ -4438,44 +4426,44 @@ public string GetQsvHwVidDecoder(EncodingJobInfo state, EncodingOptions options,
 
-            if (string.Equals(options.HardwareAccelerationType, "qsv", StringComparison.OrdinalIgnoreCase))
+            if (is8bitSwFormatsQsv)
             {
-                switch (videoStream.Codec.ToLowerInvariant())
-                {
-                    case "avc":
-                    case "h264":
-                        return _8bitSwFormatsQsv.Contains(videoStream.PixelFormat)
-                            ? GetHwaccelType(state, options, "h264", bitDepth, hwSurface) + GetHwDecoderName(options, "h264", "qsv", "h264", bitDepth)
-                            : string.Empty;
-                    case "hevc":
-                    case "h265":
-                        return _8_10bitSwFormatsQsv.Contains(videoStream.PixelFormat)
-                            ? GetHwaccelType(state, options, "hevc", bitDepth, hwSurface) + GetHwDecoderName(options, "hevc", "qsv", "hevc", bitDepth)
-                            : string.Empty;
-                    case "mpeg2video":
-                        return _8bitSwFormatsQsv.Contains(videoStream.PixelFormat)
-                            ? GetHwaccelType(state, options, "mpeg2video", bitDepth, hwSurface) + GetHwDecoderName(options, "mpeg2", "qsv", "mpeg2video", bitDepth)
-                            : string.Empty;
-                    case "vc1":
-                        return _8bitSwFormatsQsv.Contains(videoStream.PixelFormat)
-                            ? GetHwaccelType(state, options, "vc1", bitDepth, hwSurface) + GetHwDecoderName(options, "vc1", "qsv", "vc1", bitDepth)
-                            : string.Empty;
-                    case "vp8":
-                        return _8bitSwFormatsQsv.Contains(videoStream.PixelFormat)
-                            ? GetHwaccelType(state, options, "vp8", bitDepth, hwSurface) + GetHwDecoderName(options, "vp8", "qsv", "vp8", bitDepth)
-                            : string.Empty;
-                    case "vp9":
-                        return _8_10bitSwFormatsQsv.Contains(videoStream.PixelFormat)
-                            ? GetHwaccelType(state, options, "vp9", bitDepth, hwSurface) + GetHwDecoderName(options, "vp9", "qsv", "vp9", bitDepth)
-                            : string.Empty;
-                    case "av1":
-                        return _8_10bitSwFormatsQsv.Contains(videoStream.PixelFormat)
-                            ? GetHwaccelType(state, options, "av1", bitDepth, hwSurface) + GetHwDecoderName(options, "av1", "qsv", "av1", bitDepth)
-                            : string.Empty;
+                if (string.Equals(videoStream.Codec, "avc", StringComparison.OrdinalIgnoreCase)
+                    || string.Equals(videoStream.Codec, "h264", StringComparison.OrdinalIgnoreCase))
+                {
+                    return GetHwaccelType(state, options, "h264", bitDepth, hwSurface) + GetHwDecoderName(options, "h264", "qsv", "h264", bitDepth);
+                }
```

Comment:
```suggestion
                }

```

Closing brace needs a newline before starting next block.
Repeat for all if blocks here